#!/usr/bin/env python
PACKAGE = "focus"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("dtau_u", double_t, 1, "dtau_u", 0.27, 0.01, 0.5)
gen.add("dtau_v", double_t, 1, "dtau_v", 0.27, 0.01, 0.5)
gen.add("beta", double_t, 1, "beta", 1.0, 0.5, 1.5)
gen.add("sg", double_t, 2, "sg", 5.0, 0.1, 15.0)
gen.add("Ap", double_t, 2, "Ap", 1.0, 0.5, 1.5)
gen.add("sp", double_t, 2, "sp", 5.0, 0.01, 15.0)
gen.add("Am", double_t, 2, "Am", 1.0, 0.5, 1.5)
gen.add("sm", double_t, 2, "sm", 30.0, 1, 1000.0)
gen.add("circular", bool_t, 2, "Circular symmetric ?", False)
gen.add("shift_rad", double_t, 0, "max small shift radius", 0.2, 0.0, 1.0)
gen.add("shift_freeze", double_t, 0, "shift freeze duration", 0.5, 0.0, 2.0)

exit(gen.generate(PACKAGE, "focus", "Param"))
